src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -862,23 +862,24 @@ public void remove_rollback_directory_if_exists(string packageName)
 
 			foreach (var packageName in packageNames)
 			{
-				// If using local repository as source all packages have set IsLatestVersion to true.
-				var latestPackages = packageManager.SourceRepository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).ToList();
+				var repository = packageManager.SourceRepository;
 
-				IPackage latestPackage = null;
+				bool isRemote = IsRepositoryRemote(repository);
 
-				if(latestPackages.Count > 1)
+				IPackage latestPackage;
+
+				if (isRemote)
 				{
-					latestPackage = latestPackages.Last();
+					latestPackage = repository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).SingleOrDefault();

```

Comment:
If there is no package in a remote source nuget produces warning ```[NuGet] Object reference not set to an instance of an object.``` Is there a way to suppress it? If not we can call .ToList().SingleOrDefault().

Reply:
Since we are acting on that list immediately, you can `.ToList()` it. No need to defer results any further.